language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/TypeGuardUtil.java

```diff
@@ -0,0 +1,121 @@
+import io.ballerina.compiler.api.symbols.UnionTypeSymbol;
+import io.ballerina.compiler.syntax.tree.ExpressionNode;
+import io.ballerina.compiler.syntax.tree.SimpleNameReferenceNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import org.ballerinalang.langserver.SnippetBlock;
+import org.ballerinalang.langserver.common.utils.CommonUtil;
+import org.ballerinalang.langserver.commons.BallerinaCompletionContext;
+import org.ballerinalang.langserver.commons.completion.LSCompletionItem;
+import org.ballerinalang.langserver.completions.SnippetCompletionItem;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Dynamic snippet constructs for typeguards.
+ */
+public class TypeGuardUtil {
+
+    private static final String TYPE_GUARD_LABEL = "typeguard";
+
+    /**
+     * Given a union type variable symbol, returns the typeguard snippet destructuring the
+     * member types of the given symbol.
+     *
+     * @param symbol Uniontype symbol
+     * @param ctx    Completion context
+     * @return
+     */
+    public static LSCompletionItem getTypeGuardDestructedItem(UnionTypeSymbol symbol,
+                                                              BallerinaCompletionContext ctx,
+                                                              SimpleNameReferenceNode expr) {
+        String symbolName = expr.name().text();
+        List<TypeSymbol> members = new ArrayList<>(symbol.memberTypeDescriptors());
+        String detail = "Destructure the variable " + symbolName + " with typeguard";
+        String snippet = "";
+        snippet += IntStream.range(0, members.size() - 1).mapToObj(value -> {
+            TypeSymbol bType = members.get(value);
+            String placeHolder = "\t${" + (value + 1) + "}";
+            return "if " + symbolName + " is " + CommonUtil.getModifiedTypeName(ctx, bType) + " {"
+                    + CommonUtil.LINE_SEPARATOR + placeHolder + CommonUtil.LINE_SEPARATOR + "}";
+        }).collect(Collectors.joining(" else ")) + " else {" + CommonUtil.LINE_SEPARATOR + "\t${"
+                + members.size() + "}" + CommonUtil.LINE_SEPARATOR + "}";
+
+        //create the text edit item to replace the field accession text.
+        List<TextEdit> textEdits = new ArrayList<>();
+        TextEdit textEdit = new TextEdit();
+        textEdit.setNewText("");
+        Position start = new Position(expr.lineRange().startLine().line(), expr.lineRange().startLine().offset());
+        Position end = new Position(expr.lineRange().endLine().line(), expr.lineRange().endLine().offset() + 1);
+        textEdit.setRange(new Range(start, end));
+        textEdits.add(textEdit);
+
+        SnippetBlock cItemSnippet = new SnippetBlock(TYPE_GUARD_LABEL, snippet, detail,
```

Comment:
Instead of using the `SnippetBlock`, let's introduce a new CompletionItemBuilder. Refer `WorkerCompletionItemBuilder`

Reply:
Done!